test/Farmer/addProduct.test.js

```diff
@@ -0,0 +1,53 @@
+import { expect } from "chai";
+import nock from "nock";
+import { describe, it } from "mocha";
+import { BASEURL } from "../../src/core/urls.js";
+import { getUserFarmsSuccess, userFarmsAreNil } from "./farmerResponses.js";
+import categoriesFound, {
+  categoriesServerError,
+} from "./productcallResponses.js";
+import { fetchCategories } from "../../src/products/productmanagement.js";
+import { getUserFarms } from "../../src/users/farmer/farmmanagement.js";
+
+describe("Add product", () => {
+  beforeEach(() => {
+    nock(`${BASEURL}`)
+      // TODO:Backend resolve this issue
+      .get("/ussd/farm/31")
+      .reply(200, getUserFarmsSuccess);
+
+    nock(`${BASEURL}`).get("/ussd/farm/4").reply(404, userFarmsAreNil);
+    nock(`${BASEURL}`).get("/ussd/prodcategories").reply(200, categoriesFound);
```

Comment:
Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/JoeMindo/USSDTest/pullRequest?prid=8695127)
```suggestion
    nock(`${BASEURL}`).get('/ussd/prodcategories').reply(200, categoriesFound);
```